packages/react-ui-core/src/Gmap/Markers.js

```diff
@@ -90,12 +90,12 @@ export default class Markers extends PureComponent {
 
     if (!map || !geojson || !geojson.features) return
 
-    geojson.features.forEach(feature => {
+    geojson.features.forEach((feature, index) => {
       const id = feature.properties.id
 
       if (this.markers[id]) return
 
-      this.markers[id] = setupMarker(map, this.marker(feature))
+      this.markers[id] = setupMarker(map, this.marker(feature), index)
```

Comment:
So you have a set of items in `geojson`, you loop through them and save a marker based on the id, and you save the geojson `index` of the item.

What happens if `geojson` changes, but the new array contains some of the same items? It seems like the previous marker would still exist and have an `index` value that is no longer correct?

Reply:
Also does that mean that this is no longer a pure component?

Reply:
+ 1 for Fitz's first point here

Reply:
When the `geojson` changes, we render a completely new set of Markers.

I don't understand why this change would make this component "impure".

Reply:
For example:
* I call the component with geojson ids [ a, b, c ]
* It loops through and create this.markers = { a: index0, b: index1, c: index2 }
* I update the component with geojson ids [ b, d ]
* It loops through and sees that this.markers.b already exists so it does not create a new marker, and creates {d: index1}
* So would I end up with this.markers = { b: index1, d: index1 }